lib/lotus/model/migrator.rb

```diff
@@ -0,0 +1,290 @@
+      #
+      # By default, migrate to latest migration resided in db/migrations
+      #
+      # @example Migrate to the latest migration
+      #   require 'lotus/model/migration'
+      #   adapter_cfg = Lotus::Model::Config::Adapter.new(type: :sql, uri: DATABASE_URI)
+      #   Lotus::Model::Migrator.new(adapter_cfg).migrate
+      #
+      # @since 0.3.0
+      #
+      # @see Lotus::Model::Migrator#run
+      def migrate
+        options = {
+          target: nil,
+          current: nil,
+        }
+        run(options)
+      end
+
+      # Get the current schema version
+      #
+      # @since 0.3.0
+      # @api private
+      def current_version
+        last_record = _schema_migrations_dataset.last
+        last_record ? _extract_version_from(last_record[_default_schema_column]) : 0
+      end
+
+      private
+
+      # Get the migration directory
+      #
+      # @since 0.3.0
+      # @api private
+      def _migration_directory
+        configuration.migration_directory
+      end
+
+      # Get the schema version before the current current version
+      #
+      # @since 0.3.0
+      # @api private
+      def target_version(step)
+        record = _schema_migrations_dataset.all[-(1+step)]
+        record ? _extract_version_from(record[_default_schema_column]) : 0
+      end
+
+      # Extract the version from filename
+      #
+      # @example
+      #   _extract_version_from("20150222124516_create_comments.rb")
+      #   # => 20150222124516
+      #
+      # @since 0.3.0
+      # @api private
+      def _extract_version_from(filename)
+        filename.split(::Sequel::Migrator::MIGRATION_SPLITTER, 2).first.to_i
+      end
+
+      # Read data from schema_migrations table
+      #
+      # @since 0.3.0
+      # @api private
+      def _schema_migrations_dataset
+        @ds ||= db.from('schema_migrations').order(_default_schema_column)
+      end
+
+      # Get schema column
+      #
+      # @since 0.3.0
+      # @api private
+      def _default_schema_column
+        ::Sequel::TimestampMigrator::DEFAULT_SCHEMA_COLUMN
+      end
+
+      # Set logging to +Lotus::Logger+
+      #
+      # @since 0.3.0
+      # @api private
+      def _set_logger
+        db.logger = _logger
+      end
+
+      # Check to ensure adapter is configured
+      #
+      # @since 0.3.0
+      # @api private
+      def _check_if_adapter_is_configured
+        raise MissingAdapterConfiguration unless _adapter_config
+      end
+
+      # Check if adapter supports migration
+      #
+      # @since 0.3.0
+      # @api private
+      def _check_if_adapter_support_migration
+        adapter_type = _adapter_config.type
+
+        unless SUPPORTED_ADAPTERS.include?(adapter_type)
+          raise Lotus::Model::Migrator::UnsupportedAdapter.new(adapter_type)
+        end
+      end
+
+      # Return the adapter configuration
+      #
+      # @since 0.3.0
+      # @api private
+      def _adapter_config
+        configuration.adapter_config
```

Comment:
Correct if I'm wrong, but Lotus::Application.adapter coexist if Lotus::Model.config.adapter, so if we override Lotus::Model.config.adapter tha previously adapter will be lost. That's the expected behaviour ?